### ISSUE TITLE
model Buildings.Examples.ChillerPlant.DataCenterDiscreteTimeControl fails verification

### ISSUE THREAD

**Issue by AndreaBartolini** [OpenModelica/OpenModelica#12651]
The following variable of the model `Buildings.Examples.ChillerPlant.DataCenterDiscreteTimeControl` fails the verification vs the Dymola reference file:
- `pumCHW.m_flow`
![image](https://github.com/OpenModelica/OpenModelica/assets/25974985/aa202737-0c85-4ada-a5d1-33e45e865ada)

Herebelow the same variable simulated on my machine:
Win11 pro - 64 bit
Dymola Version 2024x, 2023-10-06
OpenModelica v1.24.0-dev-168-g907b8bd1e8 (64-bit)
Building master 985582a3ae50ef899ee521460ae89c3cd768bb95

![image](https://github.com/OpenModelica/OpenModelica/assets/25974985/6d63d025-0387-4966-ae87-a43f6322fd6f)

OpenModelica and Dymola results overlaps, and are according to the OpenModelica simulation in the CI.

![image](https://github.com/OpenModelica/OpenModelica/assets/25974985/43fe45dc-b105-4e1f-8c0e-c73f75a416bd)

So the conclusion is that I cannot reproduce the difference between Dymola and OpenModelica on my machine.
I'll open a ticket on the Buildings issue tracker to ask a cross check.

**Comment by mwetter** [comment]
@AndreaBartolini : Are you using the flags on the following line and the lines below:
https://github.com/lbl-srg/BuildingsPy/blob/829518dc28a3a1089824e0b33f82f78801feeb9e/buildingspy/development/regressiontest.py#L3215

Note that we update to Dymola 2024x Refresh 1 a few days ago, but this should not matter for this model as the (low resolution) reference results did not change since June 2022:
https://github.com/lbl-srg/modelica-buildings/blob/master/Buildings/Resources/ReferenceResults/Dymola/Buildings_Examples_ChillerPlant_DataCenterDiscreteTimeControl.txt

@JayHuLBL: Can you please compare the Dymola results vs the high resolution reference results that are stored at https://simulationresearch.lbl.gov/ReferenceResults/modelica-buildings/master/Dymola/ for the variable `pumCHW.m_flow`. 
Please also check whether the `.mat` file has the same number of time stamps as 
https://simulationresearch.lbl.gov/ReferenceResults/modelica-buildings/master/Dymola/Buildings_Examples_ChillerPlant_DataCenterDiscreteTimeControl.txt
I think we subsample to around 500 points before writing the `.txt` files. If this is the case, then that likely explains why for such high frequency results, the `.txt` don't verify successfully. If this is the case, we should consider storing the `.mat` outputs as written by Dymola on https://simulationresearch.lbl.gov/ReferenceResults, possibly after filtering out all non-used variables if the size would otherwise be excessive. These files are not version controlled or backed up, so having large files is less of an issue.

**Comment by AndreaBartolini** [comment]
Herebelow my setting about paralelization in Dymola:
![image](https://github.com/OpenModelica/OpenModelica/assets/25974985/0666c40f-5042-4574-95f0-44ed1a59b1ac)

It seems to me that they are according to https://github.com/lbl-srg/BuildingsPy/blob/829518dc28a3a1089824e0b33f82f78801feeb9e/buildingspy/development/regressiontest.py#L3215

**Comment by JayHuLBL** [comment]
@mwetter @AndreaBartolini 
I compared the high resolution results, with the dymola result (master branch, Dymola version 2024xRefresh1, Ubuntu 22.04, run as click `Commands -> Simulate and Plot`). They have the same number of time stamps (10567) and the difference as below:
![Picture1](https://github.com/OpenModelica/OpenModelica/assets/23491921/fc889c58-403b-408c-a063-acd05d47f831)

So the difference are in the range `[-5.0e-7, 5.0e7]`.

**Comment by AndreaBartolini** [comment]
At this point it seems that the difference is between Dymola 2024x Refresh 1 and Dymola Version 2024x, 2023-10-06 / OpenModelica v1.24.0-dev-168-g907b8bd1e8 (64-bit).

**Comment by casella** [comment]
I wonder if it really makes sense to expect that the simulation of switching systems gives exactly the same results no matter how the numerical solution is carried out. I have no theoretical background to say that it doesn't, as it is instead possible for continuous, nonlinear chaotic systems, but I wonder.

At the end of the day, the behaviour of a switching system depends on thresholds being crossed, it is likely that accumulated integration errors eventually cause some switchings to be different. The other 5 variables are fine, so the overall behaviour of the system is the same.

Maybe it's really sensible to remove this variable from the set of reference trajectories, because it is too sensitive to numerical issues.

I think a good argument to support this conclusion could be to check how this flow rate simulation changes with the same tool, as you change the tolerance from 1e-4 to 1e-8. If this were a continuous smooth system, the numerical solution would gradually approach a limit trajectory, which is the exact solution of the ODEs. But this is a switching system, so I expect to see significant differences in this m_flow. So, a tool cannot even agree with itself, depending on the tolerance. 

This argument could then be used to state that this variable is too sensitive to numerical noise for its exact values to be significant. Only its average behaviour, which BTW is corrected by feedback in this case, is significant, and this is reflected in the trajectories of the system outputs (e.g. temperatures) which happen to be accurate enough. Hence, we should remove m_flow from the reference variables.

@AndreaBartolini could you please try that and report the results here. Then we can draw some conclusions with @mwetter.

**Comment by AndreaBartolini** [comment]
Before to make the test @ different tolerances, I've installed Dymola Version 2024x Refresh 1, 2024-04-12 and the result doesn't change:
Win11 pro - 64 bit
Dymola Version 2024x Refresh 1, 2024-04-12
OpenModelica v1.24.0-dev-168-g907b8bd1e8 (64-bit)
Building master 985582a3ae50ef899ee521460ae89c3cd768bb95

![image](https://github.com/OpenModelica/OpenModelica/assets/25974985/c306be95-db9e-4d86-b5ef-a0102380fa18)

At this point the only difference should be the OS

**Comment by AndreaBartolini** [comment]
Herebelow the analysis @ different tolerances (1e-4, 1e-6, 1e-8)
![image](https://github.com/OpenModelica/OpenModelica/assets/25974985/8a479f80-8d3b-4b4c-9cfe-dcedcf608807)

Versions are:
OpenModelica v1.24.0-dev-183-g504611b14f (64-bit) - builder MinGW - Compiler Clang
Dymola 2024x Refresh 1, 2024-04-12 (64-bit) - compiler: Visual Studio 2022/ Visual C++ 2022 (17)
Sysop WIN 11 pro (64-bit)

The result changes at different tolerances in different way depending on the compiler.

Dymola and OpenModelica results overlap @ tolerances 1e-6 and 1e-8
![image](https://github.com/OpenModelica/OpenModelica/assets/25974985/b3decf1f-e008-4cf4-bf8b-2dfd57031485)

**Comment by casella** [comment]
To me the case is clear: the exact behaviour of this signal is very sensitive to numerical errors, but the physical outcome (e.g. output temperatures) isn't, since the other signals pass verification.

@mwetter, my proposal is to remove this signal from the references, because it is simply not reliable enough from a numerical point of view.

**Comment by mwetter** [comment]
I agree that the mass flow can be removed. This will be corrected through https://github.com/lbl-srg/modelica-buildings/pull/3940

**Comment by AndreaBartolini** [comment]
Solved by https://github.com/lbl-srg/modelica-buildings/pull/3940
